Firestore/Source/Remote/FSTRemoteStore.h

```diff
@@ -106,14 +106,13 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithLocalStore:(FSTLocalStore *)localStore
                          datastore:
                              (std::shared_ptr<firebase::firestore::remote::Datastore>)datastore
-                       workerQueue:(firebase::firestore::util::AsyncQueue *)queue;
+                       workerQueue:(firebase::firestore::util::AsyncQueue *)queue
+               onlineStateDelegate:(id<FSTOnlineStateDelegate> _Nullable)onlineStateDelegate;
```

Comment:
Taking the delegate as a parameter in constructor is more similar to what other platforms are doing. `_Nullable` is to avoid a warning in `FSTDatastoreTests.mm`, which pass `nil` here.